implementations/rust/ockam/ockam_api/src/nodes/models/policy.rs

```diff
@@ -39,3 +39,80 @@ impl Expression {
         &self.expr
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use quickcheck::{quickcheck, Arbitrary, Gen, TestResult};
+
+    use ockam_abac::{Action, Expr};
+
+    use crate::nodes::models::policy::{Expression, PolicyList};
+    use crate::schema::tests::validate_with_schema;
+
+    quickcheck! {
+        fn policy_list(g: PolicyList) -> TestResult {
+            validate_with_schema("policy_list", g)
+        }
+    }
+
+    quickcheck! {
+        fn policy(g: Expression) -> TestResult {
+            validate_with_schema("expression", g)
+        }
+    }
+
+    impl Arbitrary for PolicyList {
+        fn arbitrary(g: &mut Gen) -> Self {
+            PolicyList {
+                expressions: vec![
+                    Expression {
+                        action: Action::new(String::arbitrary(g).as_str()),
+                        expr: Expr::Str(String::arbitrary(g)),
+                    },
+                    Expression {
+                        action: Action::new(String::arbitrary(g).as_str()),
+                        expr: Expr::Int(i64::arbitrary(g)),
+                    },
+                    Expression {
+                        action: Action::new(String::arbitrary(g).as_str()),
+                        expr: Expr::Float(f64::arbitrary(g)),
+                    },
+                    Expression {
+                        action: Action::new(String::arbitrary(g).as_str()),
+                        expr: Expr::Bool(bool::arbitrary(g)),
+                    },
+                    Expression {
+                        action: Action::new(String::arbitrary(g).as_str()),
+                        expr: Expr::Ident(String::arbitrary(g)),
+                    },
+                    Expression {
+                        action: Action::new(String::arbitrary(g).as_str()),
+                        expr: Expr::Seq(vec![
+                            Expr::Str(String::arbitrary(g)),
+                            Expr::Str(String::arbitrary(g)),
+                        ]),
+                    },
+                    Expression {
+                        action: Action::new(String::arbitrary(g).as_str()),
+                        expr: Expr::List(vec![
+                            Expr::Int(i64::arbitrary(g)),
+                            Expr::Float(f64::arbitrary(g)),
+                        ]),
+                    },
+                ],
+            }
+        }
+    }
+
+    impl Arbitrary for Expression {
+        fn arbitrary(g: &mut Gen) -> Self {
+            Expression {
+                action: Action::new(String::arbitrary(g).as_str()),
+                expr: Expr::List(vec![
```

Comment:
I would suggest to extract an `Arbitrary for Expr` and extend it to other types of `Expr` (`Str`, `Ident`, etc...).

Reply:
same as above. Expression is in ockam_api and the rest is in ockam_abac